### PR TITLE
Visually Improve the Push Notification Authentication Step

### LINKED ISSUE #33550: The Two-Factor Authentication Step Is a Visual Dead End
Bottle Pickup

Whenever I land on this screen, it doesn’t feel actionable. It requires a few seconds to realize what’s the next step for the user that’s trying to sign in. Personally, I find it hard to recognize there’s an option to use my own authenticator app, which I log in with every single time.

I suggest we redesign this screen to:

* visually emphasize the fact that next step requires opening a notification from the mobile app.
* make it clearer that the user can use their own authenticator app.

### PR DESCRIPTION
**Pull request by porada**
#### Changes proposed in this Pull Request

When two-factor authentication is enabled:

* Visually emphasize the fact that the flow requires opening a notification from the mobile app.
* Make it clearer that the user can choose to continue with an alternative action (using their own authenticator app or by getting a code via text).

Meta:

* Add support for `<Button block />` that removes the necessity for superfluous module-specific class names.

#### Testing instructions

* Make sure two-factor authentication is enabled on your account.
* Log out if you’re signed in.
* Log in.
* When the push notification step appears:
    * Make sure it works exactly like on [the video](https://cloudup.com/coX_t6HnAoo) and looks like on the screenshot below.
    * The push notification illustration animates in.

Repeat in all the supported browsers.

Fixes #33550.

[Screenshot](https://cloudup.com/coX_t6HnAoo)

### CODE REVIEW

**Comment by porada** [comment]
Before we merge this:

@obi2020, any chance you could give me a hand editing the original piece of copy below? It feels so clunky right now.

> We sent a push notification to your **WordPress mobile app**. Once you get it and swipe or tap to confirm, this page will update.

**Comment by obi2020** [comment]
@porada What does the notification say?  I would streamline something like the following, granted the notification in the app sets the proper context.

`Confirm this login the WordPress mobile app to continue.`

`Notification sent! Confirm in your WordPress mobile app to continue.`

**Comment by porada** [comment]
> What does the notification say?

It’s pretty generic:

```
New sign in from [location] …
```

But with your suggestions, it shouldn’t really matter. They’re super clear and concise. Thank you!

**Comment by porada** [comment]
Updated the copy, updated the screenshot above, and fixed the linting issues.

Ready for review.

**Comment by donpark** [comment]
I'm getting this on Safari. I've not yet tried other browsers.

screenshot_699

File: client/blocks/login/two-factor-authentication/verification-code-form.jsx [hunk 1]
```diff
@@ -143,9 +143,9 @@ class VerificationCodeForm extends Component {
 
 					<FormFieldset>
 						<FormLabel htmlFor="twoStepCode">{ labelText }</FormLabel>
-
+						// The use of `autoFocus` is intentional in this short flow.
```
Comment (donpark) [inline client/blocks/login/two-factor-authentication/verification-code-form.jsx]: JSX comments need to be bracketed in `{` `}`.

**Comment by porada** [comment]
Whoa, I must have done this in the last commit. A proper fix is coming
later today.

File: client/blocks/login/two-factor-authentication/verification-code-form.jsx (hunk 1 above)
Comment (blowery) [inline client/blocks/login/two-factor-authentication/verification-code-form.jsx]: can't use `//` comments inside react like this
```suggestion
						{ // The use of `autoFocus` is intentional in this short flow. }
```

**Comment by blowery** [comment]
This looks amazing! Thank you so much!

File: client/blocks/login/two-factor-authentication/verification-code-form.jsx (hunk 1 above)
Comment (blowery) [reply client/blocks/login/two-factor-authentication/verification-code-form.jsx]: hah, i see Don added the same comment just before me. Sorry for the dupe.

**Comment by porada** [comment]
Fixed the last commit. Thanks for the review, @donpark and @blowery.

**Comment by donpark** [comment]
How are you testing this PR? I'm not the animation due to CSP errors on Safari and Chrome.

Safari
screenshot_707

Chrome
screenshot_710

**Comment by porada** [comment]
That’s weird. It works for me both locally and at [calypso.live](https://calypso.live/?branch=fix/33550-2fa-login-step) (in both Chrome and Safari).

**Comment by donpark** [comment]
> That’s weird. It works for me both locally and at [calypso.live](https://calypso.live/?branch=fix/33550-2fa-login-step) (in both Chrome and Safari).

Weird indeed. Calypso.live has same problem for me.

Let's see what Ben sees.

File: client/blocks/login/style.scss [hunk 2]
```diff
@@ -213,30 +214,76 @@
...
 	width: 100%;
+	height: 180px;
+	pointer-events: none;
 }
 
-.card.two-factor-authentication__form-action {
-	background: var( --color-white );
+.two-factor-authentication__illustration-device {
+	position: absolute;
+	top: 0;
```
Comment (jancavan) [inline client/blocks/login/style.scss]: We should indent position properties under a position declaration.

File: client/blocks/login/style.scss [hunk 3]
```diff
@@ -213,30 +214,76 @@
...
+	line-height: 1em;
+}
 
-		&:hover {
-			color: var( --color-accent );
-		}
-	}
+.two-factor-authentication__illustration-notification-container {
+	position: absolute;
+	top: 60px;
```
Comment (jancavan) [inline client/blocks/login/style.scss]: Indent

File: client/blocks/login/style.scss [hunk 4]
```diff
@@ -213,30 +214,76 @@
...
+	overflow: hidden;
 }
 
-.two-factor-authentication__small-print {
-	clear: both;
-	font-size: 11px;
-	padding: 1em 0;
+.two-factor-authentication__illustration-notification {
+	position: absolute;
+	top: -57px;
```
Comment (jancavan) [inline client/blocks/login/style.scss]: Also indent

File: client/blocks/login/style.scss [hunk 5]
```diff
@@ -213,30 +214,76 @@
...
+	width: 352px;
+	height: 180px;
+}
 
-	button {
-		color: var( --color-link-dark );
-		cursor: pointer;
+.two-factor-authentication__illustration-screen {
+	position: absolute;
+	top: 66px;
```
Comment (jancavan) [inline client/blocks/login/style.scss]: Indent

File: client/blocks/login/style.scss (hunk 2 above)
Comment (porada) [reply client/blocks/login/style.scss]: ✅

**Comment by jancavan** [comment]
This works great 👍

**Comment by blowery** [comment]
@donpark CSP is in report-only, it doesn't actually block anything yet. So that's probably not the problem.

Trying it on my end.

**Review by blowery — APPROVED** [review]
Works here too, thanks!

**Comment by porada** [comment]
Ship it parrot

**Pull request merged**